core/src/wallet/cosmos/database/CosmosLikeTransactionDatabaseHelper.cpp

```diff
@@ -0,0 +1,682 @@
+/*
+ *
+ * CosmosLikeTransactionDatabaseHelper
+ *
+ * Created by El Khalil Bellakrid on 06/01/2019.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+
+#include <wallet/cosmos/database/CosmosLikeTransactionDatabaseHelper.hpp>
+
+#include <database/soci-option.h>
+#include <database/soci-date.h>
+#include <database/soci-number.h>
+#include <crypto/SHA256.hpp>
+#include <wallet/common/database/BlockDatabaseHelper.h>
+
+#include <wallet/cosmos/database/SociCosmosAmount.hpp>
+#include <wallet/cosmos/CosmosLikeConstants.hpp>
+
+#include <api/enum_from_string.hpp>
+
+#include <boost/lexical_cast.hpp>
+
+#include <algorithm>
+#include <unordered_map>
+
+#include <utils/Exception.hpp>
+
+namespace ledger {
+    namespace core {
+
+        static void inflateMessage(soci::session& sql, soci::row const& row, cosmos::Message& msg) {
```

Comment:
As I mentioned in previous comment this kind of body function should be avoid. There are too many behaviours containing and the cognitive complexity is too high. 
Hopefully there are a lot of solutions to solve that problem :
- Create dedicated methods and call it in the case or if/else statement
- Create a map of functions and call it with the message type as a key
- Create helper for each message and call the appropriate method